src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java

```diff
@@ -62,15 +63,16 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         response.addResponseParam(Const.ParamsNames.CHECK_PERSISTENCE_COURSE, getStudent().course);
         excludeStudentDetailsFromResponseParams();
 
-        if (statusToAdmin == null || statusToAdmin.trim().isEmpty()) {
-            statusToAdmin = studentInfo;
+        if (statusToAdmin == null || statusToAdmin.isEmpty()) {
+            statusMessageToAdmin += studentInfo;
         } else {
-            statusToAdmin += "<br><br>" + studentInfo;
+            statusMessageToAdmin += "<br><br>" + studentInfo;
         }
```

Comment:
The original behaviour is to append to the existing status message. This approach adds a new status message to the list, which is a change in behaviour.

Reply:
This our new approach is calling statusToAdmin.add once, after the entire message has been prepared. 
Is it change in behaviour ?

Reply:
That's assuming `statusToAdmin` is only modified here. Consider the case where `statusToAdmin` is neither null nor empty upon reaching this part of the code. The new behaviour would actually result in two messages, when the original intention was for one message.

Reply:
We are not appending to `statusToAdmin` anywhere. Making ` statusMessageToAdmin` the entire message and then put it in statusToAdmin

Reply:
What's the solution to this ? :sweat_smile:

Reply:
@whipermr5 Any thought on this?